src/schema/schema_create.c

```diff
@@ -44,6 +44,49 @@ __wt_direct_io_size_check(
     return (0);
 }
 
+/*
+ * __check_imported_ts --
+ *     Check the aggregated timestamps for each checkpoint in a file that we've imported. We're not
+ *     allowed to import files with timestamps ahead of our stable timestamp since a subsequent
+ *     rollback to stable could result in data loss. Therefore, this function should return non-zero
+ *     to callers to signify that this is the case.
+ */
+static int
+__check_imported_ts(WT_SESSION_IMPL *session, const char *uri, const char *config)
+{
+    WT_CKPT *ckptbase, *ckpt;
+    WT_DECL_RET;
+    WT_TXN_GLOBAL *txn_global;
+
+    ckptbase = NULL;
+    txn_global = &S2C(session)->txn_global;
+
+    WT_ERR_NOTFOUND_OK(__wt_meta_ckptlist_get(session, uri, false, &ckptbase, config), true);
+    if (ret == WT_NOTFOUND)
+        WT_ERR_MSG(session, EINVAL,
+          "%s: import could not find any checkpoint information in supplied metadata", uri);
+
+    /* Now iterate over each checkpoint and compare the aggregate timestamps with our global. */
+    WT_CKPT_FOREACH (ckptbase, ckpt) {
+        if (ckpt->ta.newest_start_durable_ts > txn_global->stable_timestamp)
+            WT_ERR_MSG(session, EINVAL,
+              "%s: import found an aggregated durable timestamp newer than the current stable "
+              "timestamp, newest_start_durable_ts=%" PRIu64 ", stable_ts=%" PRIu64,
+              uri, ckpt->ta.newest_start_durable_ts, txn_global->stable_timestamp);
+
+        if (ckpt->ta.oldest_start_ts > txn_global->stable_timestamp)
+            WT_ERR_MSG(session, EINVAL,
+              "%s: import found an aggregated commit timestamp newer than the current stable "
+              "timestamp, oldest_start_ts=%" PRIu64 ", stable_ts=%" PRIu64,
+              uri, ckpt->ta.oldest_start_ts, txn_global->stable_timestamp);
+    }
```

Comment:
I think we also need to add checks for the stop timestamps like this:
```
if (ckpt->ta.newest_stop_ts != WT_TS_MAX && ckpt->ta.newest_stop_ts > txn_global->stable-timestamp)
```

Reply:
5e08f5c

Reply:
I think there's something wrong here. I think that in the case of stop timestamps, the aggregated time isn't enough information.

In the case that not every record has a tombstone, `newest_stop_ts` will be `WT_TS_MAX`. In that case, it's not possible whether there's any non-MAX timestamp greater than stable.

Reply:
@tetsuo-cpp I saw the above comment, I am not sure of the full details. But there is an aggregated `newest_durable_stop_ts` that stores the newest tombstone that is present on the table. Maybe it is useful for you?

Reply:
@kommiharibabu 
Maybe. If there is a record with no tombstone, will `newest_durable_stop_ts` be `WT_TS_MAX`? Or is it the newest non-max timestamp.

Reply:
The `newest_durable_stop_ts` will always be a non-max timestamp even if the record with no tombstone.

Reply:
Had a chat with @kommiharibabu.
Looks like the behaviour of `newest_durable_stop_ts` tells us what we need.